### PR TITLE
fix: updated parse_specific_ingredients subroutine

### LINKED ISSUE #3642: Add handling for "amount of product for each Xg/mL of product"
### What
- Commonly found in sauces and purees but also in meat products, an example:
  - Tomate* (160g de tomate por cada 100g de producto final), code 8425205013532
- This structure is currently unhandled and results in unknown.

Another example:
Elabora con 59 g de fruta por 100 g
Code 8410095507001

### Part of
- #5706

### PR DESCRIPTION
**Pull request by benbenben2**
### What
Updated the parse_specific_ingredients subroutine to cover more languages. See issue.

> Swedish: Fruktråvara x g per 100g  

could not find this text in any product, found other text instead

> English: 148g per 100g

is missing ingredients, @stephanegigandet if you think there is a way to handle this case - tomatoes (148g per 100g) - for example. https://world.openfoodfacts.org/product/5000157074409/heinz-tomato-ketchup-sweet-chilli

> Préparée avec 61 g de fruits pour 100g
> Teneur totale en sucres: 61 g pour 100 g.

already implemented

### Screenshot
Screenshot_20230923_132927

### Related issue(s) and discussion
- Fixes #3642

### CODE REVIEW

**Review by stephanegigandet — APPROVED** [review]
Thank you for adding support for all those languages!

It works, but it creates a lot of duplicate code.

One thing we could do is to have language-specific strings stored for things like "por cada", "de production final" etc. in regexps like =~ /\($percent_or_quantity_regexp\s*(?:de )\s*([^,.;]+?)\s*(?:por cada )100\s*g (?:de producto final)\s*\)/i

That way we could probably have only one loop for all languages (at least the ones where the order of the components is the same).

We could merge it as-is, and then refactor later if that's ok with you.

**Comment by stephanegigandet** [comment]
> > English: 148g per 100g
> 
> is missing ingredients, @stephanegigandet if you think there is a way to handle this case - tomatoes (148g per 100g) - for example. https://world.openfoodfacts.org/product/5000157074409/heinz-tomato-ketchup-sweet-chilli

@benbenben2 I made a PR to handle this case: https://github.com/openfoodfacts/openfoodfacts-server/pull/9080

**Comment by benbenben2** [comment]
Thanks for the review and parallel PR.  :+1: 

> It works, but it creates a lot of duplicate code.
> 
> One thing we could do is to have language-specific strings stored for things like "por cada", "de production final" etc. in regexps like =~ /($percent_or_quantity_regexp\s*(?:de )\s*([^,.;]+?)\s*(?:por cada )100\s_g (?:de producto final)\s_)/i
> 
> That way we could probably have only one loop for all languages (at least the ones where the order of the components is the same).
> 
> We could merge it as-is, and then refactor later if that's ok with you.

If it is fine for you, give me some time to try your suggestion in this current PR (otherwise, we will close an issue to open a new one :( )

**Comment by stephanegigandet** [comment]
> If it is fine for you, give me some time to try your suggestion in this current PR (otherwise, we will close an issue to open a new one :( )

Sure, there's no hurry. :)

File: tests/unit/expected_test_results/ingredients/fr-specific-ingredients.json [hunk 1]
```diff
@@ -37,40 +37,60 @@
...
             }
          ],
-         "percent_estimate" : 9.375,
+         "percent_estimate" : 5.20833333333333,
          "percent_max" : 25,
          "percent_min" : 0,
          "text" : "gélifiant"
+      },
+      {
+         "id" : "fr:prepares-avec-25-2-de-tomates",
```
Comment (stephanegigandet) [inline tests/unit/expected_test_results/ingredients/fr-specific-ingredients.json:61]: Those cases used to be recognized before

**Comment by stephanegigandet** [comment]
@benbenben2 thanks a lot for the PR!

I continue to build upon it and added more documentation and test cases: https://github.com/openfoodfacts/openfoodfacts-server/pull/9113

Could you take a look? I think it covers well a lot of cases now.

File: tests/unit/expected_test_results/ingredients/fr-specific-ingredients.json (hunk 1 above)
Comment (benbenben2) [reply tests/unit/expected_test_results/ingredients/fr-specific-ingredients.json:61]: Just pushed correction to handle all those cases.

Comment (benbenben2) [reply tests/unit/expected_test_results/ingredients/fr-specific-ingredients.json:61]: Just saw that you opened a new PR 

Maybe you can merge manually changes of my last commit in the current PR: https://github.com/openfoodfacts/openfoodfacts-server/pull/9060/commits/cc54fd00855f162e29c56da7560a609792f5e7a1 into your new PR. 
And close this present PR.

	fr => "(?:élaboré|Elaboré|fabriqué|préparé|produit)(?:e)?(?:s)? (?:avec|à partir)", is missing prepare (without accents) in your PR which is in the test (PREPARE AVEC 30% DE TRUC INCONNU.)

this should work too, no?
(?:élaboré|Elaboré -> (?:(?:é|e)laboré

I will review your PR later this week.

Comment (benbenben2) [reply tests/unit/expected_test_results/ingredients/fr-specific-ingredients.json:61]: I added suggestion in your PR

**Pull request closed without merging**